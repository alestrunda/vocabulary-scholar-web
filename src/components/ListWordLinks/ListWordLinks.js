import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ListWordLinks = ({ className, style, words }) => {
  return (
    <ul className={className} style={style}>
      {words.map(item => {
        return (
          <li key={item.id} className={`${className}__item`}>
            <Link to={`/word/${item.id}`} className={`${className}__link`}>
              {item.word}
            </Link>
          </li>
        );
      })}
    </ul>
  );
};

ListWordLinks.propTypes = {
  className: PropTypes.string,
  words: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      word: PropTypes.string,
    })
  ),
  style: PropTypes.object,
};

export default ListWordLinks;
