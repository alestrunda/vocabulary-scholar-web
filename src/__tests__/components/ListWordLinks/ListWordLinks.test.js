import React from 'react';
import { shallow } from 'enzyme';

import ListWordLinks from '../../../components/ListWordLinks';

it('renders correctly', () => {
  const words = [
    {
      id: 'school',
      word: 'school',
    },
    {
      id: 'car',
      word: 'car',
    },
    {
      id: 'travel',
      word: 'travel',
    },
  ];
  const root = shallow(<ListWordLinks words={words} />);
  expect(root).toMatchSnapshot();
});
