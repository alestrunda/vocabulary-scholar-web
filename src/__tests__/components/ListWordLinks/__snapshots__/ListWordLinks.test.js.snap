// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListWordLinks
    words={
      Array [
        Object {
          "id": "school",
          "word": "school",
        },
        Object {
          "id": "car",
          "word": "car",
        },
        Object {
          "id": "travel",
          "word": "travel",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="undefined__item"
        >
          <ForwardRef
            className="undefined__link"
            to="/word/school"
          >
            school
          </ForwardRef>
        </li>,
        <li
          className="undefined__item"
        >
          <ForwardRef
            className="undefined__link"
            to="/word/car"
          >
            car
          </ForwardRef>
        </li>,
        <li
          className="undefined__item"
        >
          <ForwardRef
            className="undefined__link"
            to="/word/travel"
          >
            travel
          </ForwardRef>
        </li>,
      ],
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "school",
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            className="undefined__link"
            to="/word/school"
          >
            school
          </ForwardRef>,
          "className": "undefined__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "school",
            "className": "undefined__link",
            "to": "/word/school",
          },
          "ref": null,
          "rendered": "school",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "car",
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            className="undefined__link"
            to="/word/car"
          >
            car
          </ForwardRef>,
          "className": "undefined__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "car",
            "className": "undefined__link",
            "to": "/word/car",
          },
          "ref": null,
          "rendered": "car",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "travel",
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            className="undefined__link"
            to="/word/travel"
          >
            travel
          </ForwardRef>,
          "className": "undefined__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "travel",
            "className": "undefined__link",
            "to": "/word/travel",
          },
          "ref": null,
          "rendered": "travel",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="undefined__item"
          >
            <ForwardRef
              className="undefined__link"
              to="/word/school"
            >
              school
            </ForwardRef>
          </li>,
          <li
            className="undefined__item"
          >
            <ForwardRef
              className="undefined__link"
              to="/word/car"
            >
              car
            </ForwardRef>
          </li>,
          <li
            className="undefined__item"
          >
            <ForwardRef
              className="undefined__link"
              to="/word/travel"
            >
              travel
            </ForwardRef>
          </li>,
        ],
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "school",
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="undefined__link"
              to="/word/school"
            >
              school
            </ForwardRef>,
            "className": "undefined__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "school",
              "className": "undefined__link",
              "to": "/word/school",
            },
            "ref": null,
            "rendered": "school",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "car",
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="undefined__link"
              to="/word/car"
            >
              car
            </ForwardRef>,
            "className": "undefined__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "car",
              "className": "undefined__link",
              "to": "/word/car",
            },
            "ref": null,
            "rendered": "car",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "travel",
          "nodeType": "host",
          "props": Object {
            "children": <ForwardRef
              className="undefined__link"
              to="/word/travel"
            >
              travel
            </ForwardRef>,
            "className": "undefined__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "travel",
              "className": "undefined__link",
              "to": "/word/travel",
            },
            "ref": null,
            "rendered": "travel",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            },
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
